routing/bicycle_directions.cpp

```diff
@@ -0,0 +1,185 @@
+#include "routing/bicycle_directions.hpp"
+#include "routing/car_model.hpp"
+#include "routing/router_delegate.hpp"
+#include "routing/routing_result_graph.hpp"
+#include "routing/turns_generator.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "indexer/ftypes_matcher.hpp"
+#include "indexer/index.hpp"
+#include "indexer/scales.hpp"
+
+namespace
+{
+using namespace routing;
+using namespace routing::turns;
+
+class AStarRoutingResultGraph : public IRoutingResultGraph
```

Comment:
IRoutingResult (An old name is IRoutingResultGraph) is a base clase of two classes: 
AStarRoutingResult (An old name is AStarRoutingResultGraph)
and
OSRMRoutingResult (An old name is OSRMRoutingResultGraph)

The first one is used to accumulate route info for turn generation after AStar routing. The second one is used for accumulating route info for osrm routing.